src/cli/eofwrap.py

```diff
@@ -0,0 +1,339 @@
+                    gas_limit=header.gas_limit,
+                    timestamp=header.timestamp,
+                    extra_data=header.extra_data,
+                    prev_randao=header.prev_randao,
+                    nonce=header.nonce,
+                    base_fee_per_gas=header.base_fee_per_gas,
+                    withdrawals_root=header.withdrawals_root,
+                    parent_beacon_block_root=header.parent_beacon_block_root,
+                    requests_root=header.requests_root,
+                )
+                assert not fixture_block.ommers
+                assert not fixture_block.withdrawals
+                assert not fixture_block.deposit_requests
+                assert not fixture_block.withdrawal_requests
+                assert not fixture_block.consolidation_requests
+
+                for fixture_tx in fixture_block.txs:
+                    fixture_tx_dump = fixture_tx.model_dump()
+                    fixture_tx_dump.pop("ty")
+                    fixture_tx_dump.pop("data")
+                    tx = Transaction(
+                        type=fixture_tx.ty,
+                        input=fixture_tx.data,
+                        **fixture_tx_dump,
+                    )
+                    block.txs.append(tx)
+            elif isinstance(fixture_block, InvalidFixtureBlock):
+                block = Block(
+                    rlp=fixture_block.rlp,
+                    exception=fixture_block.expect_exception,
+                )
+            else:
+                raise TypeError("not a FixtureBlock")
+
+            test.blocks.append(block)
+
+        result = test.generate(
+            request=None,  # type: ignore
+            t8n=t8n,
+            fork=CancunEIP7692,
+            fixture_format=BlockchainFixture,
+        )
+        if traces:
+            print_traces(t8n.get_traces())
+        return result
+
+    def _validate_eof(self, container: Container, metrics: bool = True) -> bool:
+        eof_parse = EOFParse()
+
+        result = eof_parse.run(input=to_hex(container))
+        actual_message = result.stdout.strip()
+        if "OK" not in actual_message:
+            if metrics:
+                _inc_counter(self.metrics[self.VALIDATION_ERRORS], actual_message)
+            return False
+
+        return True
+
+
+# `no_type_check` required because OpcodeWithOperand.opcode can be `None` when formatting as a
+# string, but here it can never be `None`.
+@no_type_check
+def wrap_code(account_code: Bytes) -> Container:
+    """
+    Wraps `account_code` into a simplest EOF container, applying some simple heuristics in
+    order to obtain a valid code section termination.
+    """
+    assert len(account_code) > 0
+
+    opcodes = process_evm_bytes(account_code, allow_unknown=True)
+
+    if not opcodes[-1].opcode.terminating:
+        opcodes.append(OpcodeWithOperands(opcode=Op.STOP))
+
+    while len(opcodes) > 1 and opcodes[-2].opcode.terminating and opcodes[-1].opcode.terminating:
```

Comment:
```suggestion
    while len(opcodes) > 1 and opcodes[-2].terminating and opcodes[-1].terminating:
```
I think it would be a bit cleaner to add:
```python
@property
def terminating(self) -> bool:
    """Whether the opcode is terminating or not"""
    return self.opcode.terminating
```
To the `OpcodeWithOperands` class in `evm_bytes.py`.